### PR TITLE
[Feature][Calling] Skip Setup Screen 

### PR DESCRIPTION
**Pull request by ShaunaSong**
## Purpose

* skip setup screen 

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

**Review by Mohtasim — DISMISSED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift [hunk 1]
```diff
@@ -12,14 +12,27 @@ public struct LocalOptions {
...
+    ///  The SkipSetup is used when we skip the setup screen
+    let skipSetupScreen: Bool?
     /// Create an instance of LocalOptions. All information in this object is only stored locally in the composite.
     /// - Parameters:
     ///    - participantViewData: The ParticipantViewData to be displayed for local participants avatar
     ///    - setupScreenViewData: The SetupScreenViewData to be used to set up views on setup screen
     public init(participantViewData: ParticipantViewData? = nil,
-                setupScreenViewData: SetupScreenViewData? = nil) {
+                setupScreenViewData: SetupScreenViewData? = nil,
+                startWithCameraOn: Bool? = false,
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:27]: all these values by default should be nil

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/EnvConfig.swift [hunk 2]
```diff
@@ -52,6 +52,9 @@ class EnvConfigSubject: ObservableObject {
     @Published var locale: Locale = SupportedLocale.en
     @Published var localeIdentifier: String = ""
     @Published var isRightToLeft: Bool = false
+    @Published var micDefault: Bool = false
```
Comment (pavelprystinka) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/EnvConfig.swift:55]: Let's keep naming consistent - startWithCameraOn format for all fields

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/StoreExtensions.swift [hunk 3]
```diff
@@ -12,10 +12,23 @@ extension Store where State == AppState, Action == AzureCommunicationUICalling.A
     static func constructStore(
         logger: Logger,
         callingService: CallingServiceProtocol,
-        displayName: String?
+        displayName: String?,
+        localOptions: LocalOptions
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/StoreExtensions.swift:16]: I think we should not pass here a whole LocalOptions. It only needs startWithCameraOn, startWithMicrophoneOn and skipSetupScreen

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/StoreExtensions.swift (hunk 3 above)
Comment (ShaunaSong) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/Redux/StoreExtensions.swift]: Updated

File: AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/EnvConfig.swift [hunk 4]
```diff
@@ -52,6 +52,9 @@ class EnvConfigSubject: ObservableObject {
     @Published var locale: Locale = SupportedLocale.en
     @Published var localeIdentifier: String = ""
     @Published var isRightToLeft: Bool = false
+    @Published var startWithMicophoneOn: Bool = false
```
Comment (pavelprystinka) [inline AzureCommunicationUI/AzureCommunicationUIDemoApp/Sources/EnvConfig.swift:55]: There is a typo.

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

**Review by Mohtasim — DISMISSED** [review]

**Review by pavelprystinka — DISMISSED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Review by Mohtasim — APPROVED** [review]

**Pull request merged**